src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -172,11 +172,20 @@ trait SlickExpressionParser extends JavaTokenParsers {
   private def workspaceAttributeFinalFunc(attributeName: String)(context: SlickExpressionContext, shouldBeNone: Option[PipeType]): ReadAction[Iterable[Attribute]] = {
     assert(shouldBeNone.isEmpty)
 
-    (for {
+    val wsIdAndAttributeQuery = for {
       workspace <- workspaceQuery.findByIdQuery(context.workspaceContext.workspaceId)
       workspaceAttribute <- workspaceAttributeQuery if workspaceAttribute.workspaceId === workspace.id
       attribute <- attributeQuery if workspaceAttribute.attributeId === attribute.id && attribute.name === attributeName
-    } yield attribute).result.map(x => Seq(attributeQuery.unmarshalAttributes(x.map((_, None))).getOrElse(attributeName, AttributeNull)))
+    } yield (workspace.id, attribute)
+
+    wsIdAndAttributeQuery.result.map { wsIdAndAttributes =>
+      // the query restricts all results to have workspace id === context.workspaceContext.workspaceId
```

Comment:
Great comment/explanation, thanks!

But couldn't restricting by workspace ID and attribute name also be accomplished in the query above?